scripts/py/gen_nextmap_maplists.py

```diff
@@ -0,0 +1,179 @@
+teamffasizes = ['2v2v2','2v2v2v2','2v2v2v2v2','2v2v2v2v2v2','2v2v2v2v2v2v2','2v2v2v2v2v2v2v2','3v3v3','3v3v3v3','3v3v3v3v3','4v4v4','4v4v4v4','5v5v5']
+#output_string = ''
+
+def get_data(input_file):
+    with open(input_file) as f:
+        contents = json.load(f)
+    
+    teamsize_dict = {}
+    ffasize_dict = {}
+    teamffasize_dict = {}
+    certified_maps = []
+    uncertified_maps = []
+    maps_1v1 = []
+  
+    for i in teamsizes:
+        teamsize_dict[i] = []
+    for i in ffasizes:
+        ffasize_dict[i] = []
+    for i in teamffasizes:
+        teamffasize_dict[i] = []
+    
+    for i in contents:
+        if not contents[i]["inPool"]:
+            continue
+        isTeam = False
+        isFfa = False
+        is1v1 = False
+        mapname = ''
+        playerCount = 0
+
+        if "springName" in contents[i].keys():
+            mapname = contents[i]["springName"]
+        if "gameType" in contents[i].keys() and "team" in contents[i]["gameType"]:
+            isTeam = True
+        if "gameType" in contents[i].keys() and "ffa" in contents[i]["gameType"]:
+            isFfa = True
+        if "gameType" in contents[i].keys() and "1v1" in contents[i]["gameType"]:
+            is1v1 = True
+        if "playerCount" in contents[i].keys():
+            playerCount = contents[i]["playerCount"]
+        #32 player ffa or other such sillyness not supported in !nextmap
+        if playerCount > 16:
+            playerCount = 16
+
+        if "startboxesSet" in contents[i].keys():
+            for j in contents[i]["startboxesSet"]:
+                if "maxPlayersPerStartbox" in contents[i]["startboxesSet"][j].keys():
+                    numberOfTeams = 0
+                    for k in contents[i]["startboxesSet"][j]["startboxes"]:
+                        numberOfTeams = numberOfTeams + 1
+                    maxPlayersPerStartbox = contents[i]["startboxesSet"][j]["maxPlayersPerStartbox"]
+
+                    # add maps to teamsize_dict
+                    if numberOfTeams == 2:
+                        x = 0
+                        for l in teamsize_dict:
+                            if x + 2 <= maxPlayersPerStartbox and x + 2 >= math.floor(maxPlayersPerStartbox/2):
+                                teamsize_dict[l].append(mapname)
+                            x = x + 1
+                    
+                    # add maps to teamffasize_dict:
+                    if numberOfTeams == 3:
+                        if maxPlayersPerStartbox >= 2:
+                            teamffasize_dict['2v2v2'].append(mapname)
+                        if maxPlayersPerStartbox >= 3:
+                            teamffasize_dict['3v3v3'].append(mapname)
+                        if maxPlayersPerStartbox >=4:
+                            teamffasize_dict['4v4v4'].append(mapname)
+                        if maxPlayersPerStartbox >=5:
+                            teamffasize_dict['5v5v5'].append(mapname)
```

Comment:
I would much more prefer if this wasn't a series of if conditions, but the strings were created dynamically.

For example:

```py
>>> teams = 3
>>> players = 2
>>> 'v'.join([str(players)] * teams)
'2v2v2'
```

Reply:
Yes your way definitely looks better! Thanks for showing it. Just had to add some sanity checks (no 1v1v1v1 or 5v5v5v5 allowed in teamffa).